mvc/views/razor.rst

```diff
@@ -1,8 +1,283 @@
+properties and method of that model via ``@Model.SomeProperty`` or
+``@Model.SomeMethod()``.
+
+.. code-block:: c#
+
+  ISBN: @Model.ISBN
+  Author: @Model.AuthorName()
+
+The ``@using`` directive is the Razor equivalent of the c# ``using`` statement.
+
+.. code-block:: c#
+
+  @using Example.Models;
+
+************
+Code Blocks
+************
+Code blocks are a fundamental element of razor. A code block is a section where
+c# code, has been inserted directly into the html. It follows the same rules
+that a normal c# block does. Implicit expressions do not have trailing
+semicolons.
+
+.. code-block:: c#
+
+  @{
+    var foo = 123;
+  }
+
+The combination of code blocks and implicit expressions allow variables to be
+created and used.
+
+.. code-block:: c#
+
+  <div id="greeting">
+    @{
+      var GuestName = "John";
+    }
+
+    <h2>Greetings @GuestName</h2>
+  </div>
+
+You can mix plain text, and html inside of a code block by using
+html tags and the ``<text>`` tag. The ``<text>`` tag is a placeholder tag that
+is only used server side to tell razor that the following content is text and
+should not be parsed as c# code.
+
+.. code-block:: c#
+
+  @{
+    var title = "Hello!";
+    <h1>@title</h1>
+
+    var body = "some text not sitting inside of a tag";
+    <text>@body</text>
+  }
+
+*******************
+Control Structures
+*******************
+A lot of the power behind razor comes from the ability to have control structures
+inside of your markup. C# control structures such as ``foreach``, ``if``,
+``using``, etc are available with Razor.
+
+.. code-block:: c#
+
+  @if(Model.IsRenderable){
+    <ul>
+    @foreach(var item in Model.Items){
+      <li>@item.Name</li>
+    }
+    </ul>
+  }
+
+
+*********
+Comments
+*********
+The Razor ``@* *@`` comment block is analogous to the c# ``/* */`` comment
+block:
+
+.. code-block:: c#
+
+  @*
+    Some comment
+  *@
+
+***************************
+Application Relative Paths
+***************************
+One of my favorite features of razor is the ability to use application relative
+paths.
+
+.. code-block:: c#
+
+  <a href="~/Home">Home page</a>
+  <script type='text/javascript' src="~/some.js">
+
+This only works for attributes that are defined in HTML5 as urls. If you want
+to resolve a path in an attribute not defined as a url you can use
+
+.. code-block:: c#
+
+  <a cust-attr="@Url.Content("~/the-url")" href="#"> Link </a>
+
+***************
+Tag Attributes
+***************
+Razor makes it easy to have programmatic classes and ids for you html tags.
+
+.. code-block:: c#
+
+  @{
+    var AnchorClass = "dark link";
+    var AnchorId = "main-link";
+  }
+
+  <a href="#" class="@AnchorClass" id="@AnchorId"> Link </a>
+
+The output for this markup is below.
+
+.. code-block:: html
+
+  <a href="#" class="dark link" id="main-link"> Link </a>
+
+When the value you pass into an attribute is null razor will remove that
```

Comment:
There's a secondary aspect to this, it also helps in pretty-printing. For instance if you have

```
<a href="#" class="bootstrap-anchor @classes"> Link </a>
```

Razor will remove the space preceding the `@classes` value in the attribute (if it's null or false) so you'd get:
```
<a href="#" class="bootstrap-anchor"> Link </a>
```